src/Chapter3.hs

```diff
@@ -479,7 +504,38 @@ After defining the city, implement the following functions:
    complicated task, walls can be built only if the city has a castle
    and at least 10 living __people__ inside in all houses of the city totally.
 -}
-
+data Castle = Castle String | NoneCastle
+data Building = Church | Library
+data House = One | Two | Three | Four
+data Wall = Wall | NoWall
+data City = City{
+  cityCastle :: Castle,
+  cityWall :: Wall,
+  cityBuilding :: Building,
+  cityHouses :: [House]
+}
+buildCastle :: City -> String -> City
+buildCastle c n = c {cityCastle = Castle n}
+
+buildHouse :: City -> House -> City
+buildHouse c h = case c of
+  City _ _ _ hs -> c {cityHouses = h:hs}
```

Comment:
No need to use `case-of` here, as there is only one constructor, so you have a few options here:

1. 
    ```haskel
    buildHouse c h = c {cityHouses = h : cityHouses c}
    ```
    or
    ```haskel
    buildHouse c h = c {cityHouses = h : hs}
      where
        hs :: [House]
        hs = cityHouses c
    ```
2. `buildHouse (City _ _ _ hs) h = c {cityHouses = h:hs}` (but I won't recommend this one)
3. There is a `RecordWildCards` extension in Haskell, that allows you to use record field names as values:
    ```haskell
    buildHouse City{..} h = c {cityHouses = h : cityHouse}
    ```
You can learn more about the extension here: https://kodimensional.dev/recordwildcards